### ISSUE TITLE
2+ snippets with the same name

### ISSUE THREAD

**Issue by TomMalbran** [brackets-userland/brackets-snippets#15]
I was thinking of the idea of having snippets use the same name, but have some way of knowing which one you want when triggering it without opening the editor.

My use case was with a 2 function snippet which I could trigger using `f params`. In js I want it to use it like this:

```
function {{$1:functionName}}() {
    {{!cursor}}
}
```

But in php I want it like this:

```
{{$1:privacy}} function {{$2:funcName}}({{!cursor}}) {

}
```

This 2 example have a different amount of parameters which could help to know which one you are referring to, but I could think of an example with the same amount of parameters, like a for in js and php:

```
for (let i = 0; i < {{$1:array}}.length; i++) {
    {{!cursor}}
}
```

```
for ($i = 0; $i < count({{$1:array}}); $i++) {
    {{!cursor}}
}
```

Since right now the trigger is the same as the file name, maybe you could add a trigger metadata, so that the snippet name could be something more descriptive and unique, but the triggers could be the same. You could then distinguish between the methods using first the language and then the amount of parameters, and if that isn't enough, you could show the editor.

**Comment by rhbecker** [comment]
> You could then distinguish between the methods using first the language

Maybe this deserves its own enhancement-type issue, but it would seemingly make sense to allow a single snippet name to be implemented in various ways to align with respective language idiosyncrasies, comparing the already existent `lang` metadata key from the snippet against the file extension in which the snippet is being used, in order to determine which implementation to load.

**Comment by zaggino** [comment]
@rhbecker , I'm open to suggestions, ideally I'd like to see at least approximate syntax that you'd expect in the snippet file in a separate issue, then I can look at how to implement it. Syntax is probably what bothers me most currently, I want to keep it as simple as possible while powerful enough. I have seen some snippets for other editors and writing a 20 line XML file for a 2 line snippet seems very wrong to me.

**Comment by rhbecker** [comment]
Just created issue #24 to manage my tangential enhancement suggestion.

**Comment by zaggino** [comment]
This is implemented in the latest version 0.6.0.

**Comment by rhbecker** [comment]
Great! Thanks!
